### PR TITLE
Add new plugin_name to DocSchema

### PR DESCRIPTION
**Pull request by sivel**
In https://github.com/ansible/ansible/commit/bec27fb4c0a40c5f8bbcf26a475704227d65ee73 we added a new `plugin_name` field to the `ansible-doc` output. This PR adds this new field to the `DocSchema` class

See https://github.com/ansible/ansible-documentation/issues/1079

### CODE REVIEW

**Comment by gotmax23** [comment]
@felixfontein, we should also consider getting rid of the `extra="forbid"` config. I don't think we need to be that strict here.

File: src/antsibull_docs/schemas/docs/base.py [hunk 1]
```diff
@@ -619,6 +619,7 @@ class DocSchema(BaseModel):
     collection: str = REQUIRED_COLLECTION_NAME_OR_EMPTY_STR_F
     description: list[str]
     name: str
+    plugin_name: str = ""
```
Comment (gotmax23) [inline src/antsibull_docs/schemas/docs/base.py:622]: I think it'd make more sense to set this to `None` if it's not part of the docs schema.

```suggestion
    plugin_name: t.Optional[str] = None
```

Comment (sivel) [reply src/antsibull_docs/schemas/docs/base.py:622]: `filename` is also the same, it's added by ansible-doc and not part of the docs schema itself, was following that as my example.

**Comment by gotmax23** [comment]
> This seems to be indicating a change to pydantic, and nothing I've done.

There was a new pydantic release yesterday that's likely the culprit. pydantic does some funky module-level `__getattr__()` stuff that sometimes confuses static analysis tools. I'm not sure why pylint thinks a module is a function...

**Comment by sivel** [comment]
I've pushed a fix for the pylint issue in this PR.  Except for now some codecov stuff...

File: src/antsibull_docs/schemas/docs/base.py (hunk 1 above)
Comment (gotmax23) [reply src/antsibull_docs/schemas/docs/base.py:622]: Okay, fair enough.

**Review by gotmax23 — APPROVED** [review]

**Comment by felixfontein** [comment]
> @felixfontein, we should also consider getting rid of the `extra="forbid"` config. I don't think we need to be that strict here.

For the docs rendering part, ok. For the linting part, we must be strict.

**Comment by felixfontein** [comment]
Looks OK, except that it's missing a changelog fragment. I can push one.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@sivel thanks for fixing this!
@gotmax23 thanks for reviewing!

**Comment by gotmax23** [comment]
Thanks everyone!

**Pull request merged**